dev/conductor/ui/lib/main.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:conductor_core/conductor_core.dart';
+import 'package:conductor_core/proto.dart' as pb;
+
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:flutter/material.dart';
+import 'package:platform/platform.dart';
+
+const String _title = 'Flutter Conductor';
+
+const LocalFileSystem _fs = LocalFileSystem();
+const LocalPlatform _platform = LocalPlatform();
+final String _stateFilePath = defaultStateFilePath(_platform);
+
+void main() {
+  final File _stateFile = _fs.file(_stateFilePath);
+  final pb.ConductorState? state =
```

Comment:
In answer to your question @Yugue, in order to use `ConductorState` as a type here, we need to import the file in which it is declared, which is part of `package:conductor_core/proto.dart'

Reply:
FYI, I implemented this refactor because understanding the way Dart code is organized is a little complicated. The two main units of code organization are *packages* and *libraries*.

- A [package](https://dart.dev/guides/libraries/create-library-packages) is defined by a directory containing a `pubspec.yaml`, and is the unit of code that can be published to pub.dev and easily consumed by third party developers (the packages here are explicitly marked `publish_to: none` because they should only be used as part of the frontends we're building here).
- By default, a Dart file defines a *library*. If you want to split a library across multiple files, you can use the `part` keyword. Libraries are the unit of encapsulation in Dart--identifier names that start with an underscore are private to the library, and cannot be referenced outside of it.
- By [convention](https://dart.dev/guides/libraries/create-library-packages#what-makes-a-library-package), all files under `//package/lib/src` is considered private, and should not be imported by other packages. Thus I created a simple `//core/lib/proto.dart` file that only exports files from `//core/lib/src/proto/*`.

The reason why I split this code into two packages is because the CLI and the Flutter Desktop app will have different dependencies and use different test runners. I actually got this all working in a single package, but I was worried you would get blocked later so I split out the desktop app into its own package that imports the `core` via a relative path.

Ideally we would probably have three packages, `conductor_ui` and `conductor_cli` that both import `conductor_core`, but that would take a lot of refactoring and probably isn't worth the work.